GTG/core/dates.py

```diff
@@ -51,274 +52,266 @@
-        elif isinstance(value, datetime.date):
-            self._real_date = value
-            self._cached_date = value
+
+    __slots__ = ['dt_value']
+
+    def __init__(self, value=None):
+        self.dt_value = None
+        if isinstance(value, (date, datetime)):
+            self.dt_value = value
         elif isinstance(value, Date):
             # Copy internal values from other Date object
-            self._real_date = value._real_date
-            self._fuzzy = value._fuzzy
-            self._cached_date = value._cached_date
-        elif isinstance(value, str) or isinstance(value, str):
-            try:
-                da_ti = datetime.datetime.strptime(value, locale_format).date()
-                self._real_date = convert_datetime_to_date(da_ti)
-                self._cached_date = self._real_date
-            except ValueError:
+            self.dt_value = value.dt_value
+        elif value in ['None', None, '']:
+            self.dt_value = NODATE
+        elif isinstance(value, str):
+            for cls in date, datetime:
                 try:
-                    # allow both locale format and ISO format
-                    da_ti = datetime.datetime.strptime(value, ISODATE).date()
-                    self._real_date = convert_datetime_to_date(da_ti)
-                    self._cached_date = self._real_date
-                except ValueError:
-                    # it must be a fuzzy date
+                    self.dt_value = cls.fromisoformat(value)
+                    break
+                except (ValueError,  # ignoring no iso format value
+                        AttributeError):  # ignoring python < 3.7
+                    pass
+            if self.dt_value is None:
+                for date_format, accuracy in DATE_FORMATS:
                     try:
-                        value = str(value.lower())
-                        self._parse_init_value(LOOKUP[value])
-                    except KeyError:
-                        raise ValueError(f"Unknown value for date: '{value}'")
-        elif isinstance(value, int):
-            self._fuzzy = value
-            self._cached_date = FUNCS[self._fuzzy]
+                        self.dt_value = datetime.strptime(value, date_format)
+                        if accuracy is Accuracy.date:
+                            self.dt_value = self.dt_value.date()
+                        break
+                    except ValueError:
+                        pass
+            if self.dt_value is None:
+                try:  # it must be a fuzzy date
+                    self.dt_value = LOOKUP[str(value).lower()]
+                except KeyError:
+                    pass
+            if self.dt_value is None:
+                raise ValueError(f"Unknown value for date: '{value}'")
+        elif value in LOOKUP:
+            self.dt_value = LOOKUP[value]
         else:
             raise ValueError(f"Unknown value for date: '{value}'")
 
+    @property
+    def accuracy(self):
```

Comment:
I also thought about that but `cached_property` implies not using `__slots__` (or to add `__dict__` to `__slots__`, which, in my opinion, kinda defeat the purpose).
Even if not for `cached_property` I could cache the accuracy value (to be compatible with python < 3.8). 
But I don't think the method is worth it. `def accuracy` first does class comparison which is fairly quick and then check on `datetime.datetime` or `datetime.date` attributes (which use `__slots__`, also fairly quick). I'm not convinced the gain brought by caching the result would justify the overhead.

Reply:
Yeah I'd thought it could be implemented like that. I looked at the code outside the diff (where it looks even better) for very rough benchmark (was like 1 or 2 seconds slower at startup, for a sample size of 1 however), I wonder if the property could directly be calculated in the constructor, adding `accuracy` to `__slots__`. They're not big methods, so it should still be nice to look at in the code.

Reply:
I implemented caching for testing purpose, the diff seems to be from 2.5us with the caching to 3.3us without it. Here is the code to test it with.
```python
from GTG.core.dates import Date
from datetime import date
import timeit
def test():
    d = Date(date.today())
    for i in range(10):
        a = d.accuracy
timeit.timeit(test, number=100000)
```
The gain seems small enough to me not to be tempted to complexify / uglify the code, does it ?